app/questionnaire/questionnaire_schema.py

```diff
@@ -475,16 +475,16 @@ def get_operands(rules: Mapping) -> Sequence:
         operands: Sequence = rules[operator]
         return operands
 
-    def _section_ids_associated_to_list_name(self, list_name: str) -> list[str]:
-        section_ids: list[str] = []
+    def _section_ids_associated_to_list_name(self, list_name: str) -> set[str]:
+        section_ids: set[str] = set()
```

Comment:
The order can matter for chained relationships, so either use list or ordered set.

Reply:
Yep I thought as much, I have a stash which has the orderedset but couldn't safely resolve all dependency breakages - I think i may revert to list for now until Progress Value Source merges to main then I can make use of orderedset